src/mplhep/plot.py

```diff
@@ -571,8 +683,320 @@ def hist2dplot(
+            )
+            opts = denom_fill_opts
+        ax.stairs(denom_unc[0], edges=edges, baseline=denom_unc[1], **opts)
+    if guide_opts is not None:
+        opts = {"linestyle": "--", "color": (0, 0, 0, 0.5), "linewidth": 1}
+        opts.update(guide_opts)
+        if clear is not False:
+            ax.axhline(1.0, **opts)
+
+    if clear:
+        ax.autoscale(axis="x", tight=True)
+        ax.set_ylim(0, None)
+
+    return ax
+
+
 #############################################
 # Utils
+def compatible(self, other):
+    """Checks if this histogram is compatible with another, i.e. they have identical binning"""
+    if len(self.axes) != len(other.axes):
+        return False
+    if set(self.axes.name) != set(other.axes.name):
+        return False
+    # if not all(d1 == d2 for d1, d2 in zip(self.dense_axes(), other.dense_axes())):
+    #     return False
+    return True
+
+
+def poisson_interval(sumw, sumw2, coverage=_coverage1sd):
+    """Frequentist coverage interval for Poisson-distributed observations
+    Parameters
+    ----------
+        sumw : numpy.ndarray
+            Sum of weights vector
+        sumw2 : numpy.ndarray
+            Sum weights squared vector
+        coverage : float, optional
+            Central coverage interval, defaults to 68%
+    Calculates the so-called 'Garwood' interval,
+    c.f. https://www.ine.pt/revstat/pdf/rs120203.pdf or
+    http://ms.mcmaster.ca/peter/s743/poissonalpha.html
+    For weighted data, this approximates the observed count by ``sumw**2/sumw2``, which
+    effectively scales the unweighted poisson interval by the average weight.
+    This may not be the optimal solution: see https://arxiv.org/pdf/1309.1287.pdf for a proper treatment.
+    When a bin is zero, the scale of the nearest nonzero bin is substituted to scale the nominal upper bound.
+    If all bins zero, a warning is generated and interval is set to ``sumw``.
+    """
+    scale = np.empty_like(sumw)
+    scale[sumw != 0] = sumw2[sumw != 0] / sumw[sumw != 0]
+    if np.sum(sumw == 0) > 0:
+        missing = np.where(sumw == 0)
+        available = np.nonzero(sumw)
+        if len(available[0]) == 0:
+            warnings.warn(
+                "All sumw are zero!  Cannot compute meaningful error bars",
+                RuntimeWarning,
+            )
+            return np.vstack([sumw, sumw])
+        nearest = sum(
+            [np.subtract.outer(d, d0) ** 2 for d, d0 in zip(available, missing)]
+        ).argmin(axis=0)
+        argnearest = tuple(dim[nearest] for dim in available)
+        scale[missing] = scale[argnearest]
+    counts = sumw / scale
+    lo = scale * scipy.stats.chi2.ppf((1 - coverage) / 2, 2 * counts) / 2.0
+    hi = scale * scipy.stats.chi2.ppf((1 + coverage) / 2, 2 * (counts + 1)) / 2.0
+    interval = np.array([lo, hi])
+    interval[interval == np.nan] = 0.0  # chi2.ppf produces nan for counts=0
+    return interval
+
+
+def normal_interval(pw, tw, pw2, tw2, coverage=_coverage1sd):
```

Comment:
These two should also go in here https://github.com/scikit-hep/mplhep/blob/master/src/mplhep/error_estimation.py